symengine/tests/basic/test_series.cpp

```diff
@@ -0,0 +1,167 @@
+#include "catch.hpp"
+#include <iostream>
+#include <chrono>
+
+#include <symengine/series.h>
+#include <symengine/mul.h>
+#include <symengine/pow.h>
+#include <symengine/dict.h>
+
+using SymEngine::UnivariateSeries;
+using SymEngine::univariate_series;
+using SymEngine::Symbol;
+using SymEngine::symbol;
+using SymEngine::Pow;
+using SymEngine::RCP;
+using SymEngine::make_rcp;
+using SymEngine::print_stack_on_segfault;
+using SymEngine::map_uint_mpz;
+using SymEngine::Basic;
+using SymEngine::one;
+using SymEngine::zero;
+using SymEngine::integer;
+using SymEngine::vec_basic_eq_perm;
+
+TEST_CASE("Constructor of UnivariateSeries", "[UnivariateSeries")
+{
+    RCP<const Symbol> x  = symbol("x");
+    map_uint_mpz adict_ = {{0, 1}, {1, 2}, {2, 1}};
+    RCP<const UnivariateSeries> P = univariate_series(x, 2, adict_);
+    REQUIRE(P->__str__() == "1 + 2*x + O(x**2)");
+
+    RCP<const UnivariateSeries> Q = UnivariateSeries::create(x, 5, {1, 0, 2, 1});
+    REQUIRE(Q->__str__() == "1 + 2*x**2 + x**3 + O(x**5)");
+}
+
+TEST_CASE("Adding two UnivariateSeries", "[UnivariateSeries")
+{
+    RCP<const Symbol> x  = symbol("x");
+    map_uint_mpz adict_ = {{0, 1}, {1, 2}, {2, 1}};
+    map_uint_mpz bdict_ = {{0, 2}, {1, 3}, {2, 4}};
+
+    const UnivariateSeries a(x, 5, std::move(adict_));
+    const UnivariateSeries b(x, 4, std::move(bdict_));
+    RCP<const UnivariateSeries> c = add_uni_series(a, b);
+    REQUIRE(c->__str__() == "3 + 5*x + 5*x**2 + O(x**4)");
```

Comment:
Instead of testing the string representation (which can change, e.g. we might want to later decide to use `^` instead of `**` etc.), you should test the actual series itself, i.e. construct a third series that is equal to the addition, and then just compare the two series directly. The same for the rest of the tests.